src/effects/effectchainslot.cpp

```diff
@@ -824,6 +845,14 @@ void EffectChainSlot::registerInputChannel(const ChannelHandleAndGroup& handle_g
     m_channelStatusMapper.setMapping(pEnableControl, handle_group.name());
     connect(pEnableControl, SIGNAL(valueChanged(double)),
             &m_channelStatusMapper, SLOT(map()));
+
+    // NOTE(Kshitij) : Already enabled input channels should be added
+    // IMPORTANT(Kshitij) : Is this enough to init?
+    if (pInfo->pEnabled->toBool()) {
+        enableForInputChannel(pInfo->handle_group);
+    } else {
+        disableForInputChannel(pInfo->handle_group);
+    }
```

Comment:
I think you could call slotChannelStatusChanged here instead of duplicating code.